### ISSUE TITLE
[strict-boolean-expressions] Issue with vue props

### ISSUE THREAD

**Issue by RebeccaStevens** [typescript-eslint/typescript-eslint#4865]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/strict-boolean-expressions": ["error"]
  }
}
```

```TS
export type Props = {
  value?: boolean;
};

export type Emits = {
  (event: "change", value: boolean): void;
};

const props = withDefaults(defineProps<Props>(), {
  value: false,
});

const emit = defineEmits<Emits>();

const onChange = () => {
  emit("change", !props.value);
               // ~~~~~~~~~~~ - eslint @typescript-eslint/strict-boolean-expressions: Unexpected any value in conditional. An explicit comparison or type cast is required.
};
```

```jsonc
{
  "compilerOptions": {
    "module": "ESNext",
    "target": "ES2018",
    "lib": [
      "DOM",
      "ESNext"
    ],
    "strict": true,
    "moduleResolution": "node",
    "strictNullChecks": true,
    "jsx": "preserve"
  }
}
```

**Expected Result**

No violations.

**Actual Result**

An ESLint violation occurs.

**Additional Info**

I've got no idea why this would be occurring and it might not be an error with this library at all but I thought it would be best to still report this somewhere.

Note: `props.value` is typed as `boolean` (not `boolean | undefined` or anything like that).

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.21.0` |
| `@typescript-eslint/parser`        | `5.21.0` |
| `TypeScript`                       | `4.6.3` |
| `ESLint`                           | `8.14.0` |
| `node`                             | `16.14.2` |

**Comment by bradzacher** [comment]
Could you please either provide a playground repro or a repro repo for this issue?

We mauntainers don't have any familiarity with Vue or how to set things up, so providing a concrete repro will help us to help you.

**Comment by RebeccaStevens** [comment]
I can't seem to recreate the issue in a minimal repo so I guess something is going wrong on my end. Got no idea what it is though.

Feel free to close this.
